src/control/cmd/daos_server/storage_scm_test.go

```diff
@@ -265,24 +421,150 @@ func TestDaosServer_resetPMem(t *testing.T) {
+			cfg: new(config.Server).WithEngines(
+				engine.NewConfig().WithIndex(0).
+					WithStorage(storage.NewTierConfig().
+						WithStorageClass(storage.ClassDcpm.String())),
+				engine.NewConfig().WithIndex(1).
+					WithStorage(storage.NewTierConfig().
+						WithStorageClass(storage.ClassDcpm.String())),
+			),
+			expErr: errors.New("unexpected state"),
+		},
+		"sock derived from config; engines with same numa": {
+			cfg: new(config.Server).WithEngines(
+				engine.NewConfig().WithIndex(0).
+					WithStorage(storage.NewTierConfig().
+						WithStorageClass(storage.ClassDcpm.String())),
+				engine.NewConfig().WithIndex(0).
+					WithStorage(storage.NewTierConfig().
+						WithStorageClass(storage.ClassDcpm.String())),
+			),
+			expCalls: []*storage.ScmPrepareRequest{
+				{
+					Reset:    true,
+					SocketID: &zero,
+				},
+			},
+			expErr: errors.New("unexpected state"),
+		},
+		"sock derived from config; engines with different numa; one using ram": {
+			cfg: new(config.Server).WithEngines(
+				engine.NewConfig().WithIndex(0).
+					WithStorage(storage.NewTierConfig().
+						WithStorageClass(storage.ClassRam.String())),
+				engine.NewConfig().WithIndex(1).
+					WithStorage(storage.NewTierConfig().
+						WithStorageClass(storage.ClassDcpm.String())),
+			),
+			expCalls: []*storage.ScmPrepareRequest{
+				{
+					Reset:    true,
+					SocketID: &one,
+				},
+			},
+			expErr: errors.New("unexpected state"),
+		},
 	} {
 		t.Run(name, func(t *testing.T) {
 			log, buf := logging.NewTestLogger(name)
 			defer test.ShowBufferOnFailure(t, buf)
 
+			mbp := bdev.NewProvider(log, nil)
+			smbc := &scm.MockBackendConfig{
+				PrepResetRes: tc.prepResp,
+				PrepResetErr: tc.prepErr,
+			}
+			msb := scm.NewMockBackend(smbc)
+			msp := scm.NewProvider(log, msb, nil)
+			scs := server.NewMockStorageControlService(log, nil, nil, msp, mbp)
+
+			if tc.cfg == nil {
+				tc.cfg = &config.Server{}
+			}
+
 			cmd := resetSCMCmd{
 				LogCmd: cmdutil.LogCmd{
 					Logger: log,
 				},
 				Force: !tc.noForce,
 			}
+			cmd.SocketID = tc.cmdSockID
+			cmd.config = tc.cfg
+			cmd.affinitySource = mockAffinitySource
+
+			err := cmd.resetPMem(scs.ScmPrepare)
+			test.CmpErr(t, tc.expErr, err)
+
+			if tc.expCalls == nil {
+				tc.expCalls = []*storage.ScmPrepareRequest{
+					{Reset: true},
+				}
+			}
 
-			mockScmPrep := func(storage.ScmPrepareRequest) (*storage.ScmPrepareResponse, error) {
-				return tc.prepResp, tc.prepErr
+			msb.RLock()
+			if len(tc.expCalls) == 0 {
+				if len(msb.ResetCalls) != 0 {
+					t.Fatalf("unexpected number of reset calls, want 0 got %d",
+						len(msb.ResetCalls))
+				}
+			} else {
+				if len(msb.ResetCalls) != 1 {
+					t.Fatalf("unexpected number of reset calls, want 1 got %d",
+						len(msb.ResetCalls))
+				}
+				if diff := cmp.Diff(*(tc.expCalls[0]), msb.ResetCalls[0]); diff != "" {
+					t.Fatalf("unexpected reset calls (-want, +got):\n%s\n", diff)
+				}
```

Comment:
Any reason not to just compare the expCalls and ResetCalls arrays directly?

Reply:
because one is an array of pointers and the other is not, I will fix that so that I can compare the arrays directly, done